pkg/kubelet/apis/podresources/server_v1.go

```diff
@@ -69,10 +69,15 @@ func (p *v1PodResourcesServer) List(ctx context.Context, req *podresourcesv1.Lis
 		pods = p.podsProvider.GetPods()
 	}
 
-	podResources := make([]*podresourcesv1.PodResources, len(pods))
+	podResources := make([]*podresourcesv1.PodResources, 0, len(pods))
 	p.devicesProvider.UpdateAllocatedDevices()
 
-	for i, pod := range pods {
+	for _, pod := range pods {
+		// Skip terminal pods (Failed or Succeeded).
+		// Terminal pods should not appear in podresources as they no longer consume resources.
+		if pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded {
+			continue
+		}
```

Comment:
why is not took into account by ` GetActivePods`?
IOW, can we get pod in terminal state from `GetActivePods`?

Reply:
xref: bc56d0e45a24b24ca9727911a891275b81bae69b

Reply:
Yeah, GetActivePods() already filters terminal pods via isAdmittedPodTerminal(). I've moved the filtering to only apply when using GetPods(), which may include terminal pods. Added comments to clarify this distinction.